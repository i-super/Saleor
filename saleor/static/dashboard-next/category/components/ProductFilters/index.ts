export { default } from "./ProductFilters";
export * from "./ProductFilters";
