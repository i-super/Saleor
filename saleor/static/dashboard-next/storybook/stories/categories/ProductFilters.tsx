import { storiesOf } from "@storybook/react";
import * as React from "react";

import ProductFilters from "../../../category/components/ProductFilters";

const productTypes = [
  { id: "123123123", name: "Type 1" },
  { id: "123123124", name: "Type 2" },
  { id: "123123125", name: "Type 3" },
  { id: "123123126", name: "Type 4" }
];
const productFilters = {
  highlighted: "false",
  name: "Lorem ipsum",
  price_max: "50",
  price_min: "30",
  productTypes: ["123123123", "123123126"],
  published: "true"
};

storiesOf("Categories / ProductFilters", module)
  .add("with initial data", () => (
    <ProductFilters
      formState={productFilters}
      handleClear={() => {}}
      handleSubmit={() => {}}
      productTypes={productTypes}
    />
  ))
  .add("without initial data", () => (
    <ProductFilters
      handleClear={() => {}}
      handleSubmit={() => {}}
      productTypes={productTypes}
    />
  ));
