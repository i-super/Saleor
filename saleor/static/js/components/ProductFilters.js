import React, { Component, PropTypes } from 'react'
import Relay from 'react-relay';
import queryString from 'query-string';


class ProductFilters extends Component {

  static propTypes = {
    attributes: PropTypes.array,
    onFilterChanged: PropTypes.func.isRequired,
  };

  constructor(props) {
    super(props);
    this.state = {
      filters: {}
    };
  }

  onClick = (attribute, value, event) => {
    const attrValue = `${attribute}:${value}`;
    let url = '';
    const element = event.target;

    this.setState({
      filters: Object.assign(
        this.state.filters,
        {[attrValue]: !this.state.filters[attrValue]})
    });
    const enabled = Object.keys(this.state.filters).filter(key => this.state.filters[key] === true);
    this.props.onFilterChanged(enabled);

    enabled.map((param, index) => {
      param = param.replace(':', '=');
      if (index == 0) {
        url += '?' + param;
      } else {
        url += '&' + param;
      }
    });

    if (enabled.length == 0) {
      url = location.href.split('?')[0];
    }

    history.pushState({}, null , url); 

    if (element.classList.contains('active')) {
      element.classList.remove('active');
    } else {
      element.classList.add('active');
    }
  }

  componentDidMount() {
    let url_params = queryString.parse(location.search);
    Object.keys(url_params).map((params) => {
      let attrValue = '';
      if (Array.isArray(url_params[params])) {
        url_params[params].map((param) => {
          attrValue = `${params}:${param}`;
          this.setState({
            filters: Object.assign(
              this.state.filters,
              {[attrValue]: true})
          });
          const elementID = 'attr'+params+param;
          document.getElementById(elementID).classList.add('active');
        })
      } else {
        attrValue = `${params}:${url_params[params]}`;
        this.setState({
          filters: Object.assign(
            this.state.filters,
            {[attrValue]: true})
        });
        const elementID = 'attr'+params+url_params[params];
        document.getElementById(elementID).classList.add('active');
      }
    })

    const enabled = Object.keys(this.state.filters).filter(key => this.state.filters[key] === true);
    this.props.onFilterChanged(enabled);
  }

  render() {
    const { attributes } = this.props;
    return (
      <div>
        {attributes && (attributes.map((attribute) => {
          return (
            <div key={attribute.id} className="attribute">
              <ul className={attribute.name}>
                <h3>{attribute.display}</h3>
                {attribute.values.map((value) => {
                  const colorStyle = {
                    backgroundColor: value.color
                  }
                  return (
                    <button
                      id={'attr' + attribute.pk+value.pk}
                      key={value.id}
                      className="item"
                      style={colorStyle}
                      onClick={(event) => this.onClick(attribute.pk, value.pk, event)}
                    >
                      {value.display}
                    </button>
                  )
                })}
              </ul>
            </div>
          )
        }))}
      </div>
    )
  }
}

export default Relay.createContainer(ProductFilters, {
  fragments: {
    attributes: () => Relay.QL`
      fragment on ProductAttributeType @relay(plural: true) {
        id
        pk
        name
        display
        values {
          id
          pk
          display
          color
        }
      }
    `,
  },
});
