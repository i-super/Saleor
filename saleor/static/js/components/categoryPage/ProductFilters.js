import React, { Component, PropTypes } from 'react';
import Relay from 'react-relay';

import AttributeInput from './AttributeInput';
import FilterHeader from './FilterHeader';

class ProductFilters extends Component {

  constructor(props) {
    super(props);
    this.state = {
      visibility: {}
    };
  }

  static propTypes = {
    attributes: PropTypes.array,
    checkedAttributes: PropTypes.array,
    onFilterChanged: PropTypes.func.isRequired
  }

  getFilterKey(attributeName, valueSlug) {
    return `${attributeName}:${valueSlug}`;
  }

  onClick = (attributeName, valueSlug) => {
    this.props.onFilterChanged(this.getFilterKey(attributeName, valueSlug));
  }

  changeVisibility = (target) => {
    this.setState({
      visibility: Object.assign(this.state.visibility, {[target]: !this.state.visibility[target]})
    });
  }

  componentWillMount() {
    this.props.attributes.map((attribute) => {
      const attrValue = `${attribute.name}`;
      if (window.innerWidth <= 700) {
        this.setState({
          visibility: Object.assign(this.state.visibility, {[attrValue]: false})
        });
      } else {
        this.setState({
          visibility: Object.assign(this.state.visibility, {[attrValue]: true})
        });
      }
    });
  }

  render() {
    const { attributes, checkedAttributes } = this.props;
    const { visibility } = this.state;
    return (
      <div className="attributes">
        {attributes && (attributes.map((attribute) => {
          return (
            <div key={attribute.id}>
              <FilterHeader
                onClick={() => this.changeVisibility(attribute.name)}
                title={attribute.display}
                visibility={visibility[attribute.name]}
              />
              <ul id={attribute.name}>
                {attribute.values.map((value) => {
                  const key = this.getFilterKey(attribute.name, value.slug);
                  const isKeyChecked = checkedAttributes.indexOf(key) > -1;
                  if (visibility[attribute.name] || isKeyChecked) {
                    return (
                      <li key={value.id} className="item">
                        <AttributeInput
                          attribute={attribute}
                          checked={isKeyChecked}
                          onClick={this.onClick}
                          value={value}
                        />
                      </li>
                    );
                  }
                })}
              </ul>
            </div>
          );
        }))}
      </div>
    );
  }
}

export default Relay.createContainer(ProductFilters, {
  fragments: {
    attributes: () => Relay.QL`
      fragment on ProductAttributeType @relay(plural: true) {
        id
        pk
        name
        display
        values {
          id
          slug
          display
          color
        }
      }
    `
  }
});
