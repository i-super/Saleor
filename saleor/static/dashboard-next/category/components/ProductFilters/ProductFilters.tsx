import TextField from "material-ui/TextField";
import * as React from "react";

import FilterCard from "../../../components/FilterCard";
import FormSpacer from "../../../components/FormSpacer";
import MultiSelectField from "../../../components/MultiSelectField";
import { PriceRangeField } from "../../../components/PriceField";
import SingleSelectField from "../../../components/SingleSelectField";
import i18n from "../../../i18n";

const { Component } = React;

interface ProductFiltersState {
  debounceTimeout?: any;
  highlighted: string;
  name: string;
  price_min: string;
  price_max: string;
  productTypes: string[];
  published: string;
}
interface ProductFiltersProps {
  productTypes: Array<{
    id: string;
    name: string;
  }>;
  formState?: {
    highlighted: string;
    name: string;
    price_min: string;
    price_max: string;
    productTypes: string[];
    published: string;
  };
  handleSubmit(formState: any);
  handleClear();
}

export class ProductFilters extends Component<
  ProductFiltersProps,
  ProductFiltersState
> {
  static defaultState = {
    highlighted: "",
    name: "",
    price_max: "",
    price_min: "",
    productTypes: [],
    published: ""
  };
  constructor(props) {
    super(props);
    this.state = { ...ProductFilters.defaultState, ...props.formState };
  }

  render() {
    const { handleSubmit, productTypes, handleClear } = this.props;
    const publishingStatuses = [
      {
        label: i18n.t("Published", { context: "Product publishing status" }),
        value: "1"
      },
      {
        label: i18n.t("Not published", {
          context: "Product publishing status"
        }),
        value: "0"
      },
      {
        label: i18n.t("All", {
          context: "Product publishing status"
        }),
        value: ""
      }
    ];
    const highlightingStatuses = [
      {
        label: i18n.t("Highlighted", {
          context: "Product highlighting status"
        }),
        value: "1"
      },
      {
        label: i18n.t("Not highlighted", {
          context: "Product highlighting status"
        }),
        value: "0"
      },
      {
        label: i18n.t("All", {
          context: "Product highlighting status"
        }),
        value: ""
      }
    ];

    const handleInputChange = event => {
      const { name, value } = event.target;
      const debounceTimeout = setTimeout(() => {
        handleSubmit({
          formData: { [name]: value }
        });
      }, 500);
      clearTimeout(this.state.debounceTimeout);
      this.setState({
        [name]: value,
        debounceTimeout
      });
    };

    const handleClearButtonClick = () => {
      this.setState(ProductFilters.defaultState);
      handleClear();
    };

    return (
      <FilterCard handleClear={handleClearButtonClick}>
        <TextField
          fullWidth
          name="name"
          label={i18n.t("Name", {
            context: "Product filter field label"
          })}
          onChange={handleInputChange}
          value={this.state.name || ""}
        />
        <FormSpacer />
        <MultiSelectField
          label={i18n.t("Product type", {
            context: "Product filter field label"
          })}
          choices={productTypes.map(type => ({
            label: type.name,
            value: type.id
          }))}
          name="productTypes"
          onChange={handleInputChange}
          value={this.state.productTypes || []}
        />
        <FormSpacer />
        <PriceRangeField
          currencySymbol="USD"
          label={i18n.t("Price")}
          name="price"
          onChange={handleInputChange}
          value={{
            max: this.state.price_max || "",
            min: this.state.price_min
          }}
        />
        <FormSpacer />
        <SingleSelectField
          label={i18n.t("Published", {
            context: "Product filter field label"
          })}
          choices={publishingStatuses}
          name="published"
          onChange={handleInputChange}
          value={this.state.published || ""}
        />
        <FormSpacer />
        <SingleSelectField
          label={i18n.t("Highlighted", {
            context: "Product filter field label"
          })}
          choices={highlightingStatuses}
          name="highlighted"
          onChange={handleInputChange}
          value={this.state.highlighted || ""}
        />
      </FilterCard>
    );
  }
}

export default ProductFilters;
