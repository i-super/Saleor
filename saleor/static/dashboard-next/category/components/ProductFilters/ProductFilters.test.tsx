import * as React from "react";
import * as renderer from "react-test-renderer";

import ProductFilters from "./";

const productTypes = [
  { id: "123123123", name: "Type 1" },
  { id: "123123124", name: "Type 2" },
  { id: "123123125", name: "Type 3" },
  { id: "123123126", name: "Type 4" }
];
const productFilters = {
  highlighted: "false",
  name: "Lorem ipsum",
  price_max: "50",
  price_min: "30",
  productTypes: ["123123123", "123123126"],
  published: "true"
};

describe("<ProductFilters />", () => {
  it("renders without initial state", () => {
    const component = renderer.create(
      <ProductFilters
        handleClear={jest.fn()}
        handleSubmit={jest.fn()}
        productTypes={productTypes}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders with initial state", () => {
    const component = renderer.create(
      <ProductFilters
        formState={productFilters}
        handleClear={jest.fn()}
        handleSubmit={jest.fn()}
        productTypes={productTypes}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
