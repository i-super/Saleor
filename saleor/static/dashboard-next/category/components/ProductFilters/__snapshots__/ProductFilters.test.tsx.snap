// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductFilters /> renders with initial state 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2"
>
  <form>
    <div
      className="MuiCardHeader-root-28"
    >
      <div
        className="MuiCardHeader-content-31"
      >
        <span
          className="MuiTypography-root-34 MuiTypography-headline-39 MuiCardHeader-title-32"
        >
          Filters
        </span>
      </div>
      <div
        className="MuiCardHeader-action-30"
      >
        <button
          className="MuiButtonBase-root-64 MuiIconButton-root-58"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-63"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-67"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-73"
          />
        </button>
      </div>
    </div>
    <div
      className="MuiCardContent-root-80"
    >
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-81 MuiFormControl-fullWidth-84"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-91 MuiInputLabel-root-85 MuiInputLabel-formControl-86 MuiInputLabel-animated-89 MuiInputLabel-shrink-88"
          data-shrink={true}
          htmlFor={undefined}
        >
          Name
        </label>
        <div
          className="MuiInput-root-97 MuiInput-fullWidth-104 MuiInput-formControl-98 MuiInput-underline-101"
        >
          <input
            aria-invalid={false}
            aria-required={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-105"
            defaultValue={undefined}
            disabled={false}
            id={undefined}
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="text"
            value="Lorem ipsum"
          />
        </div>
      </div>
      <div
        className="Component-spacer-111"
      />
      <div
        className="MuiFormControl-root-81 Component-formControl-114"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-91 MuiInputLabel-root-85 MuiInputLabel-formControl-86 MuiInputLabel-animated-89 MuiInputLabel-shrink-88"
          data-shrink={true}
        >
          Product type
        </label>
        <div
          className="MuiInput-root-97 MuiInput-fullWidth-104 MuiInput-formControl-98 MuiInput-underline-101"
        >
          <div
            className="MuiSelect-root-115"
          >
            <div
              aria-haspopup="true"
              aria-owns={null}
              aria-pressed="false"
              className="MuiSelect-select-116 MuiSelect-selectMenu-117 MuiInput-input-105"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="Component-chipContainer-113"
              >
                <div
                  className="MuiChip-root-132 Component-chip-112"
                  onClick={undefined}
                  onKeyDown={[Function]}
                  role="button"
                  tabIndex={-1}
                >
                  <span
                    className="MuiChip-label-137"
                  >
                    Type 1
                  </span>
                </div>
                <div
                  className="MuiChip-root-132 Component-chip-112"
                  onClick={undefined}
                  onKeyDown={[Function]}
                  role="button"
                  tabIndex={-1}
                >
                  <span
                    className="MuiChip-label-137"
                  >
                    Type 4
                  </span>
                </div>
              </div>
            </div>
            <input
              aria-invalid={false}
              aria-required={false}
              autoComplete={undefined}
              autoFocus={undefined}
              defaultValue={undefined}
              id={undefined}
              name="productTypes"
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={undefined}
              rows={undefined}
              type="hidden"
              value="123123123,123123126"
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-67 MuiSelect-icon-119"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="Component-spacer-111"
      />
      <div
        className="Component-widgetContainer-127"
      >
        <div
          className="Component-inputContainer-124"
        >
          <div
            className="MuiFormControl-root-81"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <label
              className="MuiFormLabel-root-91 MuiInputLabel-root-85 MuiInputLabel-formControl-86 MuiInputLabel-animated-89 MuiInputLabel-shrink-88"
              data-shrink={true}
              htmlFor="price_min"
            >
              Price
            </label>
            <div
              className="MuiInput-root-97 MuiInput-fullWidth-104 MuiInput-formControl-98 MuiInput-underline-101"
            >
              <input
                aria-invalid={false}
                aria-required={false}
                autoComplete={undefined}
                autoFocus={undefined}
                className="MuiInput-input-105 MuiInput-inputType-109"
                defaultValue={undefined}
                disabled={false}
                id={undefined}
                name="price_min"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={undefined}
                onKeyUp={undefined}
                placeholder={undefined}
                readOnly={undefined}
                required={undefined}
                rows={undefined}
                type="number"
                value="30"
              />
              <div
                className="MuiInputAdornment-root-128 MuiInputAdornment-positionEnd-130"
              >
                <p
                  className="MuiTypography-root-34 MuiTypography-body1-43 MuiTypography-colorTextSecondary-56"
                >
                  USD
                </p>
              </div>
            </div>
          </div>
          <span
            className="Component-separator-126"
          >
            -
          </span>
          <div
            className="MuiFormControl-root-81"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <div
              className="MuiInput-root-97 MuiInput-fullWidth-104 MuiInput-formControl-98 MuiInput-underline-101 Component-pullDown-125"
            >
              <input
                aria-invalid={false}
                aria-required={false}
                autoComplete={undefined}
                autoFocus={undefined}
                className="MuiInput-input-105 MuiInput-inputType-109"
                defaultValue={undefined}
                disabled={false}
                id={undefined}
                name="price_max"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={undefined}
                onKeyUp={undefined}
                placeholder={undefined}
                readOnly={undefined}
                required={undefined}
                rows={undefined}
                type="number"
                value="50"
              />
              <div
                className="MuiInputAdornment-root-128 MuiInputAdornment-positionEnd-130"
              >
                <p
                  className="MuiTypography-root-34 MuiTypography-body1-43 MuiTypography-colorTextSecondary-56"
                >
                  USD
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="Component-spacer-111"
      />
      <div
        className="MuiFormControl-root-81 Component-formControl-131"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-91 MuiInputLabel-root-85 MuiInputLabel-formControl-86 MuiInputLabel-animated-89 MuiInputLabel-shrink-88"
          data-shrink={true}
        >
          Published
        </label>
        <div
          className="MuiInput-root-97 MuiInput-fullWidth-104 MuiInput-formControl-98 MuiInput-underline-101"
        >
          <div
            className="MuiSelect-root-115"
          >
            <div
              aria-haspopup="true"
              aria-owns={null}
              aria-pressed="false"
              className="MuiSelect-select-116 MuiSelect-selectMenu-117 MuiInput-input-105"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203",
                  }
                }
              />
            </div>
            <input
              aria-invalid={false}
              aria-required={false}
              autoComplete={undefined}
              autoFocus={undefined}
              defaultValue={undefined}
              id={undefined}
              name="published"
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={undefined}
              rows={undefined}
              type="hidden"
              value="true"
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-67 MuiSelect-icon-119"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="Component-spacer-111"
      />
      <div
        className="MuiFormControl-root-81 Component-formControl-131"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-91 MuiInputLabel-root-85 MuiInputLabel-formControl-86 MuiInputLabel-animated-89 MuiInputLabel-shrink-88"
          data-shrink={true}
        >
          Highlighted
        </label>
        <div
          className="MuiInput-root-97 MuiInput-fullWidth-104 MuiInput-formControl-98 MuiInput-underline-101"
        >
          <div
            className="MuiSelect-root-115"
          >
            <div
              aria-haspopup="true"
              aria-owns={null}
              aria-pressed="false"
              className="MuiSelect-select-116 MuiSelect-selectMenu-117 MuiInput-input-105"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203",
                  }
                }
              />
            </div>
            <input
              aria-invalid={false}
              aria-required={false}
              autoComplete={undefined}
              autoFocus={undefined}
              defaultValue={undefined}
              id={undefined}
              name="highlighted"
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={undefined}
              rows={undefined}
              type="hidden"
              value="false"
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-67 MuiSelect-icon-119"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`<ProductFilters /> renders without initial state 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2"
>
  <form>
    <div
      className="MuiCardHeader-root-28"
    >
      <div
        className="MuiCardHeader-content-31"
      >
        <span
          className="MuiTypography-root-34 MuiTypography-headline-39 MuiCardHeader-title-32"
        >
          Filters
        </span>
      </div>
      <div
        className="MuiCardHeader-action-30"
      >
        <button
          className="MuiButtonBase-root-64 MuiIconButton-root-58"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-63"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-67"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-73"
          />
        </button>
      </div>
    </div>
    <div
      className="MuiCardContent-root-80"
    >
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-81 MuiFormControl-fullWidth-84"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-91 MuiInputLabel-root-85 MuiInputLabel-formControl-86 MuiInputLabel-animated-89"
          data-shrink={false}
          htmlFor={undefined}
        >
          Name
        </label>
        <div
          className="MuiInput-root-97 MuiInput-fullWidth-104 MuiInput-formControl-98 MuiInput-underline-101"
        >
          <input
            aria-invalid={false}
            aria-required={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-105"
            defaultValue={undefined}
            disabled={false}
            id={undefined}
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="Component-spacer-111"
      />
      <div
        className="MuiFormControl-root-81 Component-formControl-114"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-91 MuiInputLabel-root-85 MuiInputLabel-formControl-86 MuiInputLabel-animated-89"
          data-shrink={false}
        >
          Product type
        </label>
        <div
          className="MuiInput-root-97 MuiInput-fullWidth-104 MuiInput-formControl-98 MuiInput-underline-101"
        >
          <div
            className="MuiSelect-root-115"
          >
            <div
              aria-haspopup="true"
              aria-owns={null}
              aria-pressed="false"
              className="MuiSelect-select-116 MuiSelect-selectMenu-117 MuiInput-input-105"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203",
                  }
                }
              />
            </div>
            <input
              aria-invalid={false}
              aria-required={false}
              autoComplete={undefined}
              autoFocus={undefined}
              defaultValue={undefined}
              id={undefined}
              name="productTypes"
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={undefined}
              rows={undefined}
              type="hidden"
              value=""
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-67 MuiSelect-icon-119"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="Component-spacer-111"
      />
      <div
        className="Component-widgetContainer-127"
      >
        <div
          className="Component-inputContainer-124"
        >
          <div
            className="MuiFormControl-root-81"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <label
              className="MuiFormLabel-root-91 MuiInputLabel-root-85 MuiInputLabel-formControl-86 MuiInputLabel-animated-89"
              data-shrink={false}
              htmlFor="price_min"
            >
              Price
            </label>
            <div
              className="MuiInput-root-97 MuiInput-fullWidth-104 MuiInput-formControl-98 MuiInput-underline-101"
            >
              <input
                aria-invalid={false}
                aria-required={false}
                autoComplete={undefined}
                autoFocus={undefined}
                className="MuiInput-input-105 MuiInput-inputType-109"
                defaultValue={undefined}
                disabled={false}
                id={undefined}
                name="price_min"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={undefined}
                onKeyUp={undefined}
                placeholder={undefined}
                readOnly={undefined}
                required={undefined}
                rows={undefined}
                type="number"
                value=""
              />
              <div
                className="MuiInputAdornment-root-128 MuiInputAdornment-positionEnd-130"
              >
                <p
                  className="MuiTypography-root-34 MuiTypography-body1-43 MuiTypography-colorTextSecondary-56"
                >
                  USD
                </p>
              </div>
            </div>
          </div>
          <span
            className="Component-separator-126"
          >
            -
          </span>
          <div
            className="MuiFormControl-root-81"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <div
              className="MuiInput-root-97 MuiInput-fullWidth-104 MuiInput-formControl-98 MuiInput-underline-101 Component-pullDown-125"
            >
              <input
                aria-invalid={false}
                aria-required={false}
                autoComplete={undefined}
                autoFocus={undefined}
                className="MuiInput-input-105 MuiInput-inputType-109"
                defaultValue={undefined}
                disabled={false}
                id={undefined}
                name="price_max"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={undefined}
                onKeyUp={undefined}
                placeholder={undefined}
                readOnly={undefined}
                required={undefined}
                rows={undefined}
                type="number"
                value=""
              />
              <div
                className="MuiInputAdornment-root-128 MuiInputAdornment-positionEnd-130"
              >
                <p
                  className="MuiTypography-root-34 MuiTypography-body1-43 MuiTypography-colorTextSecondary-56"
                >
                  USD
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="Component-spacer-111"
      />
      <div
        className="MuiFormControl-root-81 Component-formControl-131"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-91 MuiInputLabel-root-85 MuiInputLabel-formControl-86 MuiInputLabel-animated-89"
          data-shrink={false}
        >
          Published
        </label>
        <div
          className="MuiInput-root-97 MuiInput-fullWidth-104 MuiInput-formControl-98 MuiInput-underline-101"
        >
          <div
            className="MuiSelect-root-115"
          >
            <div
              aria-haspopup="true"
              aria-owns={null}
              aria-pressed="false"
              className="MuiSelect-select-116 MuiSelect-selectMenu-117 MuiInput-input-105"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203",
                  }
                }
              />
            </div>
            <input
              aria-invalid={false}
              aria-required={false}
              autoComplete={undefined}
              autoFocus={undefined}
              defaultValue={undefined}
              id={undefined}
              name="published"
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={undefined}
              rows={undefined}
              type="hidden"
              value=""
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-67 MuiSelect-icon-119"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="Component-spacer-111"
      />
      <div
        className="MuiFormControl-root-81 Component-formControl-131"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-91 MuiInputLabel-root-85 MuiInputLabel-formControl-86 MuiInputLabel-animated-89"
          data-shrink={false}
        >
          Highlighted
        </label>
        <div
          className="MuiInput-root-97 MuiInput-fullWidth-104 MuiInput-formControl-98 MuiInput-underline-101"
        >
          <div
            className="MuiSelect-root-115"
          >
            <div
              aria-haspopup="true"
              aria-owns={null}
              aria-pressed="false"
              className="MuiSelect-select-116 MuiSelect-selectMenu-117 MuiInput-input-105"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203",
                  }
                }
              />
            </div>
            <input
              aria-invalid={false}
              aria-required={false}
              autoComplete={undefined}
              autoFocus={undefined}
              defaultValue={undefined}
              id={undefined}
              name="highlighted"
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={undefined}
              rows={undefined}
              type="hidden"
              value=""
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-67 MuiSelect-icon-119"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
